### ISSUE TITLE
Final forbidden semicolon in frontmatter ignored

### ISSUE THREAD

**Issue by igorakkerman** [ota-meshi/eslint-plugin-astro#165]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.33.0

### What version of `eslint-plugin-astro` are you using?

0.23.0

### What did you do?

<details>
<summary>A final semicolon in the frontmatter of an `*.astro` file is not flagged as a warning despite forbidden by a rule.</summary>
This is the eslint configuration file.
```jsonc
{
  "root": true,
  "parser": "@typescript-eslint/parser",
  "plugins": [
    "@typescript-eslint"
  ],
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:astro/recommended",
  ],
  "overrides": [
    {
      "files": [
        "*.astro"
      ],
      "parser": "astro-eslint-parser",
      "parserOptions": {
        "parser": "@typescript-eslint/parser",
        "extraFileExtensions": [
          ".astro"
        ],
      },
      "rules": {
        // override/add rules settings here, such as:
        // "astro/no-set-html-directive": "error"
      },
    },
  ],
  "rules": {
    "quotes": [
      "warn",
      "single"
    ],
    "semi": [
      "warn",
      "never"
    ],
    "@typescript-eslint/ban-ts-comment": [
      "error",
      {
        "ts-ignore": "allow-with-description"
      },
    ],
    "@typescript-eslint/no-non-null-assertion": "off",
  },
}
```
Please find the Astro file below.
</details>

### What did you expect to happen?

Running `eslint` should show 2 warnings about both semicolons and `eslint --fix` should remove them.

### What actually happened?

With `eslint` only one warning is shown and `eslint --fix` only removes the first semicolon.

```powershell
pnpm lint
```
```
> @example/minimal@0.0.1 lint C:\Lokal\DYNAMITE\dynamite-cv\astro-web
> eslint .


...\astro-web\src\pages\semi.astro
  2:61  warning  Extra semicolon  semi

✖ 1 problem (0 errors, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.
```

### Link to Minimal Reproducible Example

```astro
---
console.log('This semicolon is flagged as a warning ->')
console.log('This one isn\'t 🤷‍♂️ ->');
---
<html><body>Don't look up!</body></html>
```

**Comment by ota-meshi** [comment]
duplicate of #102
